units/patch_intent.go

```diff
@@ -345,12 +345,17 @@ func (j *patchIntentProcessor) buildCliPatchDoc(ctx context.Context, patchDoc *p
 	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
 
-	_, err = thirdparty.GetCommitEvent(ctx, githubOauthToken, projectRef.Owner,
+	commit, err := thirdparty.GetCommitEvent(ctx, githubOauthToken, projectRef.Owner,
 		projectRef.Repo, patchDoc.Githash)
 	if err != nil {
 		return errors.Wrapf(err, "could not find base revision '%s' for project '%s'",
 			patchDoc.Githash, projectRef.Identifier)
 	}
+	// With `evergreen patch-file`, a user can pass a branch name or tag instead of a hash. We
+	// must normalize this to a hash before storing the patch doc.
+	if commit.SHA != nil && patchDoc.Githash != *commit.SHA {
```

Comment:
I think if the `patch.Githash` is not a SHA GitHub's API will not return the commit.
```
> curl -H "Authorization: token MY_TOKEN" https://api.github.com/repos/evergreen-ci/evergreen/git/commits/master
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/git/commits/#get-a-commit"
}
```

Reply:
I got a response from `curl -i -H 'Authorization: token TOKEN' https://api.github.com/repos/evergreen-ci/evergreen/commits/master`. I think you have an extra "git". https://developer.github.com/v3/repos/commits/#get-a-single-commit

Reply:
Yes, you're correct.